flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonToRowDataConverters.java

```diff
@@ -298,15 +300,25 @@ private JsonToRowDataConverter createArrayConverter(ArrayType arrayType) {
 		};
 	}
 
-	private JsonToRowDataConverter createMapConverter(MapType mapType) {
-		LogicalType keyType = mapType.getKeyType();
+	private JsonToRowDataConverter createMapConverter(LogicalType type) {
+		LogicalType keyType;
+		LogicalType valueType;
+		if (type instanceof MapType) {
+			MapType mapType = (MapType) type;
+			keyType = mapType.getKeyType();
+			valueType = mapType.getValueType();
+		} else {
+			MultisetType multisetType = (MultisetType) type;
+			keyType = multisetType.getElementType();
+			valueType = new IntType();
+		}
```

Comment:
I would suggest to change the method parameters to `createMapConverter(String typeSummary, LogicalType keyType, LogicalType valueType)` and move the above code out of the method (i.e. move to the switch case block). This can make the method `createMapConverter` more generic.

Reply:
Good idea. Updated.